src/webgpu/api/operation/command_buffer/image_copy.spec.ts

```diff
@@ -1133,3 +1277,82 @@ g.test('copy_into_stencil_aspect')
   `
   )
   .unimplemented();
+
+g.test('write_into_stencil_aspect')
```

Comment:
Sorry this is a big change, but wouldn't there be a way to add stencil8 to the kWorkingFormats and make the texture check use the new method you made, such that all the tests will automatically start testing stencil8?

Reply:
Sorry I may not fully understand your comment here. In fact ```stencil8``` is included in ```kDepthStencilFormats``` and is tested in this test.

Reply:
Ah sorry I wasn't clear. In the file we have a lot of tests exercising many options of copies: bytesPerRow, rowsPerImage, sub copies, copies at an offset, etc. It would be really nice to find a way to "just" add the stencil copy tests to the parametrization of existing tests so that we get all the amazing coverage.

Reply:
Oh the reason I prefer adding a separated test is that the copies with depth/stencil formats are a bit different as we can only do full subresource copies with depth/stencil formats and the way to verify the result is totally different from other copies. What do you think?

Reply:
@kainino0x FYI this thread raises interesting points about helpers not handling the notion of aspects as consistently as they should.

Long term I think it would be better to have the tests be generic on the aspects etc, but that can be left for later. The solution you outlined with similar looking tests for various aspects being grouped together sounds good. (although it may be better to use `.expand(commonOffsetsAndSizeParamExpander)`). To make code duplication of code minimal, do you think we could have a helper like `uploadTextureAndVerifyCopy` but for depth / stencil?

All of the above said, I think we can land this to make forward progress first, then add the rest of the tests. This is a good step.

Reply:
The aspects problem is an issue with both the helpers and the capabilities tables. Agree we definitely need to fix it when fixing the helpers.

Also agree with using simple local helpers (just blocks of code extracted out into functions) to reduce code duplication, but it's not critical. Jiawei and I discussed this a bit over chat in the past few days as well.

> code duplication of code

"code duplication," i.e., duplication of the word "code" 👍 🤣

Reply:
Thanks for your understanding, @kainino0x and @Kangz!

I plan to consider extracting the common code paths and better organizing all the buffer-to-texture copy tests after adding b2t-stencil-copy, t2b-stencil-copy and t2b-depth-copy.

Reply:
Hi @Kangz and @kainino0x, I've updated the PR. PTAL, thanks!